### PR TITLE
Rename seedu.address.model classes

### PR DESCRIPTION
**Pull request by yongping827**
Partially address #38. 

The following classes, including all related fields and methods, are renamed as:
- Person → Transaction
- Name → Name
- Phone → Amount
- Tags → Category
- Email → Date

### CODE REVIEW

**Review by ianyong — CHANGES_REQUESTED** [review]
Update the package names too to `ay2021s1-cs2103-w16-3.finesse` as discussed.

Edit: My bad, hyphens are illegal characters. How about `ay2021s1_cs2103_w16_3.finesse` as recommended [here](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html)?

File: LICENSE [hunk 1]
```diff
@@ -2,11 +2,11 @@ MIT License
 
 Copyright (c) 2016 Software Engineering Education - FOSS Resources
 
-Permission is hereby granted, free of charge, to any person obtaining a copy
+Permission is hereby granted, free of charge, to any transaction obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the Software is
```
Comment (wltan) [inline LICENSE:9]: These don't look like intentional changes :/

File: docs/AboutUs.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ title: About Us
 
 We are a team based in the [School of Computing, National University of Singapore](http://www.comp.nus.edu.sg).
 
-You can reach us at the email <ay2021s1-cs2103t-w16-3@googlegroups.com>
+You can reach us at the date <ay2021s1-cs2103t-w16-3@googlegroups.com>
```
Comment (wltan) [inline docs/AboutUs.md:8]: Should also revert this change as well

**Review by wltan — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -81,44 +81,44 @@ public static Address parseAddress(String address) throws ParseException {
...
 
     /**
-     * Parses a {@code String email} into an {@code Email}.
+     * Parses a {@code String email} into an {@code Date}.
      * Leading and trailing whitespaces will be trimmed.
      *
      * @throws ParseException if the given {@code email} is invalid.
      */
-    public static Email parseEmail(String email) throws ParseException {
+    public static Date parseEmail(String email) throws ParseException {
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:89]: `parseEmail` should be renamed to `parseDate` with the corresponding parameter renamed. Same for all other instances.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -81,44 +81,44 @@ public static Address parseAddress(String address) throws ParseException {
...
 
     /**
-     * Parses a {@code String tag} into a {@code Tag}.
+     * Parses a {@code String tag} into a {@code Category}.
      * Leading and trailing whitespaces will be trimmed.
      *
      * @throws ParseException if the given {@code tag} is invalid.
      */
-    public static Tag parseTag(String tag) throws ParseException {
+    public static Category parseTag(String tag) throws ParseException {
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:104]: `parseTag` should be renamed to `parseCategory` with the corresponding parameter renamed. Same for all other instances.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 5]
```diff
@@ -51,18 +51,18 @@ public static Name parseName(String name) throws ParseException {
...
 
     /**
-     * Parses a {@code String phone} into a {@code Phone}.
+     * Parses a {@code String phone} into a {@code Amount}.
      * Leading and trailing whitespaces will be trimmed.
      *
      * @throws ParseException if the given {@code phone} is invalid.
      */
-    public static Phone parsePhone(String phone) throws ParseException {
+    public static Amount parsePhone(String phone) throws ParseException {
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:59]: `parsePhone` should be renamed to `parseAmount` with the corresponding parameter renamed. Same for all other instances.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 6]
```diff
@@ -51,18 +51,18 @@ public static Name parseName(String name) throws ParseException {
     }
 
     /**
-     * Parses a {@code String phone} into a {@code Phone}.
+     * Parses a {@code String phone} into a {@code Amount}.
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:54]:
```suggestion
     * Parses an {@code String amount} into a {@code Amount}.
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 7]
```diff
@@ -51,18 +51,18 @@ public static Name parseName(String name) throws ParseException {
...
      * Leading and trailing whitespaces will be trimmed.
      *
      * @throws ParseException if the given {@code phone} is invalid.
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:57]:
```suggestion
     * @throws ParseException if the given {@code amount} is invalid.
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 8]
```diff
@@ -81,44 +81,44 @@ public static Address parseAddress(String address) throws ParseException {
     }
 
     /**
-     * Parses a {@code String email} into an {@code Email}.
+     * Parses a {@code String email} into an {@code Date}.
      * Leading and trailing whitespaces will be trimmed.
      *
      * @throws ParseException if the given {@code email} is invalid.
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:87]:
```suggestion
     * @throws ParseException if the given {@code date} is invalid.
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 9]
```diff
@@ -81,44 +81,44 @@ public static Address parseAddress(String address) throws ParseException {
...
+        if (!Date.isValidEmail(trimmedEmail)) {
+            throw new ParseException(Date.MESSAGE_CONSTRAINTS);
         }
-        return new Email(trimmedEmail);
+        return new Date(trimmedEmail);
     }
 
     /**
-     * Parses a {@code String tag} into a {@code Tag}.
+     * Parses a {@code String tag} into a {@code Category}.
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:99]:
```suggestion
     * Parses a {@code String category} into a {@code Category}.
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 10]
```diff
@@ -81,44 +81,44 @@ public static Address parseAddress(String address) throws ParseException {
...
      * Leading and trailing whitespaces will be trimmed.
      *
      * @throws ParseException if the given {@code tag} is invalid.
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:102]:
```suggestion
     * @throws ParseException if the given {@code category} is invalid.
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 11]
```diff
@@ -81,44 +81,44 @@ public static Address parseAddress(String address) throws ParseException {
...
+        if (!Category.isValidTagName(trimmedTag)) {
+            throw new ParseException(Category.MESSAGE_CONSTRAINTS);
         }
-        return new Tag(trimmedTag);
+        return new Category(trimmedTag);
     }
 
     /**
-     * Parses {@code Collection<String> tags} into a {@code Set<Tag>}.
+     * Parses {@code Collection<String> tags} into a {@code Set<Category>}.
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:114]:
```suggestion
     * Parses {@code Collection<String> categories} into a {@code Set<Category>}.
```

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 12]
```diff
@@ -81,44 +81,44 @@ public static Address parseAddress(String address) throws ParseException {
...
      */
-    public static Set<Tag> parseTags(Collection<String> tags) throws ParseException {
+    public static Set<Category> parseTags(Collection<String> tags) throws ParseException {
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:116]: `parseTags` should be renamed to `parseCategories` with the corresponding parameter renamed. Same for all other instances.

**Review by ianyong — CHANGES_REQUESTED** [review]

**Comment by wltan** [comment]
> Update the package names too to `ay2021s1-cs2103-w16-3.finesse` as discussed.
> 
> Edit: My bad, hyphens are illegal characters. How about `ay2021s1_cs2103_w16_3.finesse` as recommended [here](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html)?

Recommend doing this in a separate PR. Refactoring the package would generate a large diff and make it difficult to review the model renaming.

File: src/main/java/seedu/address/model/person/Date.java [hunk 13]
```diff
@@ -4,10 +4,10 @@
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
 /**
- * Represents a Person's email in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidEmail(String)}
+ * Represents a Transaction's email in the address book.
```
Comment (zhaojj2209) [inline src/main/java/seedu/address/model/person/Date.java:7]:
```suggestion
 * Represents a Transaction's date in the address book.
```

File: src/main/java/seedu/address/model/person/Date.java [hunk 14]
```diff
@@ -30,20 +30,20 @@
     public final String value;
 
     /**
-     * Constructs an {@code Email}.
+     * Constructs an {@code Date}.
      *
      * @param email A valid email address.
      */
-    public Email(String email) {
+    public Date(String email) {
```
Comment (zhaojj2209) [inline src/main/java/seedu/address/model/person/Date.java:37]: The argument name here should probably be changed from `email` to `date`. The JavaDoc should also be changed accordingly.

File: src/main/java/seedu/address/model/person/Date.java [hunk 15]
```diff
@@ -30,20 +30,20 @@
...
         requireNonNull(email);
-        checkArgument(isValidEmail(email), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidDate(email), MESSAGE_CONSTRAINTS);
         value = email;
     }
 
     /**
      * Returns if a given string is a valid email.
```
Comment (zhaojj2209) [inline src/main/java/seedu/address/model/person/Date.java:44]:
```suggestion
     * Returns if a given string is a valid date.
```

File: src/main/resources/view/MainWindow.fxml [hunk 16]
```diff
@@ -46,7 +46,7 @@
           </padding>
         </StackPane>
 
-        <VBox fx:id="personList" styleClass="pane-with-border" minWidth="340" prefWidth="340" VBox.vgrow="ALWAYS">
+        <VBox fx:id="transactionList" styleClass="pane-with-border" minWidth="340" prefWidth="340" VBox.vgrow="ALWAYS">
           <padding>
             <Insets top="10" right="10" bottom="10" left="10" />
           </padding>
```
Comment (zhaojj2209) [inline src/main/resources/view/MainWindow.fxml:52]: In line 53, `"personListPanelPlaceholder"` should be changed to `"transactionListPanelPlaceholder"`.

**Review by zhaojj2209 — CHANGES_REQUESTED** [review]
The UI currently encounters NullPointerExceptions when the application starts up; to resolve, the following files need to be refactored.

PersonListCard.fxml: Rename to TransactionListCard.fxml
PersonListPanel.fxml: Rename to TransactionListPanel.fxml, with a change in line 7
MainWindow.fxml: Change in line 53

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -64,7 +64,7 @@ The sections below give more details of each component.
 **API** :
 [`Ui.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/Ui.java)
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `TransacionListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
```
Comment (wltan) [inline docs/DeveloperGuide.md:67]: Minor typo:
```suggestion
The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `TransactionListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
```

**Review by wltan — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 18]
```diff
@@ -42,41 +41,43 @@ public EditCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
         }
 
-        EditPersonDescriptor editPersonDescriptor = new EditPersonDescriptor();
+        EditCommand.EditTransactionDescriptor editTransactionDescriptor = new EditCommand.EditTransactionDescriptor();
```
Comment (ianyong) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:44]: Could you import `EditCommand.EditTransactionDescriptor` instead?

**Review by ianyong — CHANGES_REQUESTED** [review]

**Review by ianyong — APPROVED** [review]
LGTM!

**Review by wltan — APPROVED** [review]
LGTM!

Pitest report: https://ay2021s1-cs2103t-w16-3.github.io/reports/pitest/202010012150/

**Pull request merged**